api/api.php

```diff
@@ -158,6 +158,10 @@ function _civicrm_api_replace_variables(&$params, &$parentResult, $separator = '
           $count = count($stringParts);
         }
       }
+      // CRM-16168 If we have failed to swap it out we should unset it rather than leave the placeholder.
+      if (substr($params[$field], 0, 6) == '$value') {
```

Comment:
This produces a PHP warning when the value has been set to an array - as it can be done by the previous `while` loop:

    PHP Warning:  substr() expects parameter 1 to be string, array given in ...

Should it check again that `$params[$field]` is still a string? Tell me if I should open a new issue *just* for that...